components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -662,3 +752,164 @@ func (g *rtkSerial) Close(ctx context.Context) error {
 	}
 	return nil
 }
+
+// sendGGAMessage sends GGA messages to the NTRIP Caster over a TCP connection
+// to get the NTRIP steam when the mount point is a Virtual Reference Station.
+func (g *rtkSerial) sendGGAMessage() error {
+	g.ntripMu.Lock()
+	defer g.ntripMu.Unlock()
+
+	if !g.isConnected {
+		g.readerWriter = g.connectToVirtualBase()
+	}
+
+	// read from the socket until we know if a successful connection has been
+	// established.
+	for {
+		if g.readerWriter.Reader == nil || g.readerWriter.Writer == nil {
+			break
+		}
+		line, _, err := g.readerWriter.ReadLine()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				g.logger.Debug("EOF encountered. sending GGA message again")
+				g.readerWriter = nil
+				g.isConnected = false
+				return err
+			}
+			g.logger.Error("Failed to read server response:", err)
+			return err
+		}
+
+		if strings.HasPrefix(string(line), "HTTP/1.1 ") {
+			if strings.Contains(string(line), "200 OK") {
+				g.isConnected = true
+				break
+			} else {
+				g.logger.Error("Bad HTTP response")
+				g.isConnected = false
+				return err
+			}
+		}
+	}
+
+	ggaMessage, err := g.getGGAMessage()
```

Comment:
Calling `getGGAMessage()` from within `sendGGAMessage()` seems confusing to me: getting it sounds like we're receiving it from the server, and sending it sounds like the server is receiving it from us. Is one of them misnamed? If not, put in more comments explaining what's going on.